server/controller/src/main/resources/mapper/RuntimeVersionMapper.xml

```diff
@@ -78,39 +94,62 @@
     limit 1
   </select>
 
-  <insert id="revertTo">
+  <select id="findByVersionOrderAndRuntimeId" resultMap="runtimeVersionResultMap">
+    <if test="runtimeId != null">
+      and rv.runtime_id = #{runtimeId}
+    </if>
+    <include refid="select_runtime_version"/>
+    and rv.version_order = #{versionOrder}
+  </select>
+
+  <update id="revertTo">
     update runtime_version
-    set version_order = (select max from ((select max(version_order) as max from runtime_version))
-    as v) + 1
+    set version_order = if((select version_order
```

Comment:
how about introduce a boolean column called `head` as a pointer to deal with the reverting issue. We will have less complex SQL. The `versionOrder` is optional then.